docs/cookbook/custom-promotion-action.rst

```diff
@@ -0,0 +1,147 @@
+    {
+        const TYPE = 'cheapest_item_discount';
+
+        /**
+         * @var ProportionalIntegerDistributorInterface
+         */
+        private $proportionalDistributor;
+
+        /**
+         * @var UnitsPromotionAdjustmentsApplicatorInterface
+         */
+        private $unitsPromotionAdjustmentsApplicator;
+
+        /**
+         * @param ProportionalIntegerDistributorInterface $proportionalIntegerDistributor
+         * @param UnitsPromotionAdjustmentsApplicatorInterface $unitsPromotionAdjustmentsApplicator
+         */
+        public function __construct(
+            ProportionalIntegerDistributorInterface $proportionalIntegerDistributor,
+            UnitsPromotionAdjustmentsApplicatorInterface $unitsPromotionAdjustmentsApplicator
+        ) {
+            $this->proportionalDistributor = $proportionalIntegerDistributor;
+            $this->unitsPromotionAdjustmentsApplicator = $unitsPromotionAdjustmentsApplicator;
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function execute(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion)
+        {
+            if (!$subject instanceof OrderInterface) {
+                throw new UnexpectedTypeException($subject, OrderInterface::class);
+            }
+
+            $items = $subject->getItems();
+
+            $cheapestItem = $items->first();
+
+            $itemsTotals = [];
+
+            foreach ($items as $item) {
+                $itemsTotals[] = $item->getTotal();
+
+                $cheapestItem = ($item->getVariant()->getPrice() < $cheapestItem->getVariant()->getPrice()) ? $item : $cheapestItem;
+            }
+
+            $splitPromotion = $this->proportionalDistributor->distribute($itemsTotals, -1 * $cheapestItem->getVariant()->getPrice());
+            $this->unitsPromotionAdjustmentsApplicator->apply($subject, $promotion, $splitPromotion);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getConfigurationFormType()
+        {
+            return 'app_promotion_action_cheapest_product_discount_configuration';
+        }
+    }
+
+Register the action as a service
+--------------------------------
+
+In the ``AppBundle/Resources/config/services.yml`` configure:
+
+.. code-block:: yaml
+
+    # AppBundle/Resources/config/services.yml
+    app.promotion_action.cheapest_product_discount:
+        class: AppBundle\Promotion\Action\CheapestProductDiscountPromotionActionCommand
+        arguments: ['@sylius.proportional_integer_distributor', '@sylius.promotion.units_promotion_adjustments_applicator']
+        tags:
+            - { name: sylius.promotion_action, type: cheapest_product_discount, label: Cheapest product discount }
+
+Prepare a configuration form type for the admin panel
+-----------------------------------------------------
+
+The new action needs a form type to be available in the admin panel, while creating a new promotion.
```

Comment:
We should mention that this form type is empty because we do not have any configuration here, but if we add some fields - they will appear in admin UI.